static/app/components/onboardingWizard/newSidebar.tsx

```diff
@@ -59,14 +55,94 @@ function groupTasksByCompletion(tasks: OnboardingTask[]) {
   };
 }
 
-interface TaskProps extends Pick<OnboardingTaskStatus, 'status'> {
+interface TaskCardProps {
+  description: React.ReactNode;
+  status: 'complete' | 'inProgress' | 'skipped' | 'pending';
+  title: string;
+  action?: React.ReactNode;
+  iconTooltipText?: string;
+  onClick?: (e: React.MouseEvent) => void;
+  progress?: number;
+}
+
+function TaskCard({
+  description,
+  iconTooltipText,
+  status,
+  progress,
+  title,
+  action,
+  onClick,
+}: TaskCardProps) {
+  const theme = useTheme();
+
+  const progressValue = progress ?? 0;
+
+  return (
+    <TaskCardWrapper
+      role={onClick ? 'button' : undefined}
+      onClick={onClick}
+      hasProgress={progressValue > 0}
+    >
+      {onClick && <InteractionStateLayer />}
+      <TaskCardIcon>
+        <Tooltip
+          title={iconTooltipText}
+          disabled={!iconTooltipText}
+          containerDisplayMode="flex"
+        >
+          {status === 'complete' ? (
+            <IconCheckmark
+              css={css`
+                color: ${theme.success};
+                height: ${theme.fontSizeLarge};
+                width: ${theme.fontSizeLarge};
+              `}
+              isCircled
+            />
+          ) : status === 'skipped' ? (
+            <IconNot
+              css={css`
+                color: ${theme.disabled};
+                height: ${theme.fontSizeLarge};
+                width: ${theme.fontSizeLarge};
+              `}
+            />
+          ) : status === 'pending' ? (
+            <IconSync // TODO(Telemetry): Remove pending status
+              css={css`
+                color: ${theme.pink400};
+                height: ${theme.fontSizeLarge};
+                width: ${theme.fontSizeLarge};
+              `}
+            />
```

Comment:
this will require backend changes too and will be done in a follow-up

Reply:
after an internal discussion, we concluded that we will keep the pending status. Users can always go to the task and skip it